front-end/studio/src/app/pages/apis/{apiId}/_components/import-components.wizard.ts

```diff
@@ -289,28 +304,37 @@ export class ImportComponentsWizard {
     }
 
     private resourceTableRows(): DataTableRow[] {
-        return this.allApis.map(api => {
+        return this.allResources.map(apiOrArtifact => {
             return {
-                value: api,
+                value: apiOrArtifact,
                 cells: [
-                    { displayName: api.name },
-                    { displayName: api.description },
-                    { displayName: api.type },
-                    { displayName: moment(api.createdOn).fromNow() }
+                    { displayName: apiOrArtifact.name },
+                    { displayName: apiOrArtifact.description },
+                    { displayName: apiOrArtifact.type },
+                    { displayName: moment(apiOrArtifact.createdOn).fromNow() }
                 ]
             };
         });
     }
 
     private loadComponents(): void {
         this.components = [];
-        console.debug("[ImportComponentsWizard] Loading components from %o resources.", this.selectedApis.length);
-        let promises: Promise<ApiDefinition>[] = this.selectedApis.map( api => {
-            console.debug("[ImportComponentsWizard] Loading api definition for: ", api.name);
-            return this.apis.getApiDefinition(api.id).then(apiDef => {
-                this.processLoadedApiDef(apiDef);
-                return apiDef;
-            });
+        console.debug("[ImportComponentsWizard] Loading components from %o resources.", this.selectedResources.length);
+        let promises: Promise<ApiDefinition | ArtifactDefinition>[] = this.selectedResources.map(apiOrArtifact => {
+            // TODO find a better way to route the processing flow (instanceof does not work)
```

Comment:
Agreed - probably need to annotate the apis and artifacts with some sort of routing property.  Something like `__resourceType` which would be either `API` or `ARTIFACT`.

Reply:
This can be done in their respective Promises perhaps?